library/swift/src/Envoy.swift

```diff
@@ -15,12 +15,20 @@ public final class Envoy: NSObject {
     return self.runner.isFinished
   }
 
-  /// Initialize a new Envoy instance.
+  /// Initialize a new Envoy instance using a typed configuration.
   ///
-  /// - parameter config:   Configuration file that is recognizable by Envoy (YAML).
+  /// - parameter config:   Configuration to use for starting Envoy.
   /// - parameter logLevel: Log level to use for this instance.
-  public init(config: String, logLevel: LogLevel = .info) {
-    self.runner = RunnerThread(config: config, engine: self.engine, logLevel: logLevel)
+  public convenience init(config: Configuration = Configuration(), logLevel: LogLevel = .info) throws {
+    self.init(configFile: try config.build(), logLevel: logLevel)
+  }
+
+  /// Initialize a new Envoy instance using a string configuration.
+  ///
+  /// - parameter configFile: Configuration file to use for starting Envoy.
+  /// - parameter logLevel:   Log level to use for this instance.
+  public init(configFile: String, logLevel: LogLevel = .info) {
```

Comment:
nit: let's maybe call this `configYaml` instead of `configFile`, since there's more than one type of config file that can be used to run envoy. (Also this matches the option name that Envoy uses internally, for consistency.)

Reply:
sgtm